### ISSUE TITLE
🗄️ - Upgrade LiteDB

### ISSUE THREAD

**Issue by petrspelos** [control-net/Miunie#310]
> :warning: This issue is currently blocked until #308 is resolved.

## Summary

Our `Miunie.Infrastructure` project is currently using `LiteDB` version `4.1.4`. There has been a major release since we implemented this. The goal is to upgrade to the new version of `LiteDB` and ensure that everything still works.

## How to implement

Bumping the NuGet package version number is a good start. You will, however, need to resolve potential issues in `PersistentStorage.cs`.

## Interested? 😊 

If you'd like to help us by solving this issue, just say so in the comments. ❤️ 
And don't shy away from asking any kind of question you might have.

Should you have any questions at all, feel free to contact either me (@petrspelos) or @DraxCodes on our Discord server:

[![Discord](https://img.shields.io/discord/377879473158356992?color=7289DA&label=Discord%20Server&style=for-the-badge)](https://discord.gg/cGhEZuk)

_And of course, thank you so much for contributing! 🎉 It's because of amazing people like yourself that Miunie exists._

**Comment by AlexGipp** [comment]
Could I try to solve that one?

**Comment by petrspelos** [comment]
Of course, @AlexGipp. It's all yours. ❤️ 
Let me know if you have any questions or get stuck. 😊

**Comment by AlexGipp** [comment]
How do I start the bot so I can test if my changes cause any errors? I didn't find anything in the documentation.

**Comment by petrspelos** [comment]
@AlexGipp, you're right. The documentation is still lacking since we separated the front-ends into their own repositories recently.

It would be neat if you could perhaps write a couple of unit tests to ensure this behavior works as expected. You are going to need to ensure the generated `Miunie.db` file is deleted after each test, since unit tests should not depend on one another.
This can be achieved by your test method implementing the `IDisposable` interface and then deleting the database file in the `IDisposable.Dispose` method.

However, you're still going to want to run the application. For that, I would recommend the [`Miunie.Console`](https://github.com/control-net/miunie-console) front-end.

The setup is slightly complex, since you already started development in your own fork, I assume.

1. You should clone the Miunie console project recursively:

```bash
git clone --recurse-submodules -j8 git@github.com:control-net/miunie-console.git
```

2. You then need to build the core submodule at least once, this is triggered with the project build from a command line:

```bash
dotnet build miunie-console/src
```

3. Navigate into the Core submodule:

```bash
cd miunie-console/src/Miunie
```

Notice that `git remote -v` here suggests you're working on the main Core repository (`https://github.com/control-net/Miunie`).

You want to switch this to your fork:

You should also make sure you don't have any pending changes before you start changing the remote.

```bash
git remote set-url origin git@github.com:petrspelos/Miunie.git
```

> ⚠️ Notice my username in the url, this points to my fork.

4. Add upstream:

```bash
git remote add upstream https://github.com/control-net/Miunie
```

performing `git remote -v` should now output:
```
origin  git@github.com:YOUR-USERNAME-HERE/Miunie.git (fetch)
origin  git@github.com:YOUR-USERNAME-HERE/Miunie.git (push)
upstream        https://github.com/control-net/Miunie (fetch)
upstream        https://github.com/control-net/Miunie (push)
```

5. Ensure your fork is up to date:

```bash
git checkout master
git fetch upstream
git merge upstream/master
git push
```

At this point, you're working on an updated fork within the context for a Miunie Console submodule.

Now is when you make changes in the submodule.

To ensure you know where to work, you can perform `explorer .` to open Windows explorer there or `.\src\Miunie.sln` to directly open the solution.

> ℹ️ If you still have a previous version of this repository, you may delete it.

To run the application, you then go to the MiunieConsole.sln and run the ConsoleApp project.
The structure looks like this:

```
miunie-console
└───src
    ├───Miunie
    │   ├───.github
    │   ├───.vscode
    │   ├───docs
    │   └───src
    │       ├───Miunie.Core
    │       ├───Miunie.Discord
    │       ├───Miunie.Infrastructure
    │       ├───Miunie.InversionOfControl
    │       └───Miunie.Tests
    ├───Miunie.ConsoleApp
    └───MiunieConsole.sln
```

It's basically a repository in a repository. It's called a git submodule.

**Comment by AlexGipp** [comment]
Awesome thanks for that detailed explanation, will look into it 👍
